test/e2e/cypress/integration/clusters_overview.js

```diff
@@ -0,0 +1,306 @@
+      it('should show health status of the entire cluster of 9 hosts with partial pagination', () => {
+        cy.reloadList('clusters', 10);
+        cy.get('.health-container .health-passing').should('contain', 1);
+      });
+      it('should show health status of all 9 clusters', () => {
+        cy.reloadList('clusters', 100);
+        cy.get('.health-container .health-passing').should('contain', 1);
+        cy.get('.health-container .health-warning').should('contain', 1);
+        cy.get('.health-container .health-critical').should('contain', 1);
+      });
+    });
+  });
+
+  describe('Clusters Tagging', () => {
+    before(() => {
+      cy.get('body').then(($body) => {
+        const deleteTag = '.tn-cluster-tags x';
+        if ($body.find(deleteTag).length > 0) {
+          cy.get(deleteTag).then(($deleteTag) =>
+            cy.wrap($deleteTag).click({ multiple: true })
+          );
+        }
+      });
+    });
+    const clustersByMatchingPattern = (pattern) => (clusterName) =>
+      clusterName.includes(pattern);
+    const taggingRules = [
+      ['hana_cluster_1', 'env1'],
+      ['hana_cluster_2', 'env2'],
+      ['hana_cluster_3', 'env3'],
+    ];
+
+    taggingRules.forEach(([pattern, tag]) => {
+      describe(`Add tag '${tag}' to all clusters with '${pattern}' in the cluster name`, () => {
+        availableClusters
+          .filter(clustersByMatchingPattern(pattern))
+          .forEach((clusterName) => {
+            it(`should tag cluster '${clusterName}'`, () => {
+              cy.get(
+                `#cluster-${clusterIdByName(
+                  clusterName
+                )} > .tn-cluster-tags > .tagify`
+              )
+                .type(tag)
+                .trigger('change');
+            });
+          });
+      });
+    });
+  });
+
+  describe('Filtering the Clusters overview', () => {
+    before(() => {
+      cy.reloadList('clusters', 100);
+    });
+
+    const resetFilter = (option) => {
+      cy.intercept('GET', `/clusters?per_page=100`).as('resetFilter');
+      cy.get(option).click();
+      cy.wait('@resetFilter');
+    };
+
+    describe('Filtering by health', () => {
+      before(() => {
+        cy.get('.tn-filters > :nth-child(2) > .btn').click();
+      });
+      const healthScenarios = [
+        ['passing', 1],
+        ['warning', 1],
+        ['critical', 1],
+      ];
+      healthScenarios.forEach(
+        ([health, expectedClustersWithThisHealth], index) => {
+          it(`should show ${
+            expectedClustersWithThisHealth || 'an empty list of'
+          } clusters when filtering by health '${health}'`, () => {
+            cy.intercept('GET', `/clusters?per_page=100&health=${health}`).as(
+              'filterByHealthStatus'
+            );
+            const selectedOption = `#bs-select-1-${index}`;
+            cy.get(selectedOption).click();
+            cy.wait('@filterByHealthStatus').then(() => {
+              expectedClustersWithThisHealth == 0 &&
```

Comment:
Why this `== 0` and `> 0` comparisons?
It is not enough to check the `expectedClustersWithThisHealth` value?
In fact, this `expectedClustersWithThisHealth` will always be `1` , or the value we have in the map. We have control over it. I wouldn't do this strange checkings, it might create some false positives

Reply:
In fact, you do so in the next `Filtering by SAP system` test. It looks cleaner and easier to understand